packages/teams-js/src/public/geoLocation.ts

```diff
@@ -0,0 +1,120 @@
+import { sendAndHandleSdkError as sendAndHandleError } from '../internal/communication';
+import { ensureInitialized } from '../internal/internalAPIs';
+import { errorNotSupportedOnPlatform, FrameContexts } from './constants';
+import { DevicePermission, ErrorCode } from './interfaces';
+import { runtime } from './runtime';
+
+/**
+ * Namespace to interact with the geoLocation module-specific part of the SDK. This is the newwer version of location module.
+ */
+export namespace geoLocation {
+  /**
+   * Data struture to represent the location information
+   */
+  export interface Location {
+    /**
+      Latitude of the location
+      */
+    latitude: number;
+    /**
+      Longitude of the location
+      */
+    longitude: number;
+    /**
+      Accuracy of the coordinates captured
+      */
+    accuracy?: number;
+    /**
+      Time stamp when the location was captured
+      */
+    timestamp?: number;
+  }
+  /**
+   * Fetches current user coordinates
+   * @returns Promise of user's current {@linkcode Location}
```

Comment:
Did you mean `@link` here? (*@linkcode* doesn't seem to be a valid [typedoc tag](https://typedoc.org/guides/tags/)). The rendered doc will actually show the Promise return type and link to Location based on the function signature alone, so adding a link in the description isn't necessary (but you can if you like). For example, see [getContext()](https://docs.microsoft.com/en-us/javascript/api/@microsoft/teams-js/app?view=msteams-client-js-latest#@microsoft-teams-js-app-getcontext) documentation (the broken link in the rendered doc is a known issue under investigation).

Reply:
Also for return description, could we modify slightly to match previous function descriptions that we promis-ified? For example, `@returns Promise that will resolve with the {@link location.Location} object` or `@returns Promise that will resolve with information about the user's location`.